# Report1

### Import library and read data

In [70]:
# import library
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

#read data
df = pd.read_excel('/Users/riva5/Downloads/data_query-tbl-details.xlsx')

Firstly, we import pandas as a tool and read the excel file 'data_query-tbl-details' to start analysis.

## I. Types of methodology

### Number of Different Types of Methodology

In [40]:
list(df.columns.values)
df_count = dt[['Sectoral scopes','Methodologies Used']]
df_count['Methodologies Used'] = df_count['Methodologies Used'].str[:3]

df_count['Methodologies Used'].value_counts()

ACM    4330
AMS    3254
AM0     520
AR-      67
Name: Methodologies Used, dtype: int64

We use list() to get the name of each column and take out the first three letters of 'Methodologies Used' to classify methodologies - ACM, AMS, or AM for each project.

According to the result, there are 4330 projects using ACM methodology, 3254 projects using AMS methodology, 520 projects using AM methodology, and 67 projects using methodology starting with 'AR'.


## II. Project classification methods- Methodology and Project Number

### The datafame for the number of projects with different Industry CDM scope and Methodologies.

In [44]:
# Remove AR- and revise 'AM0' to 'AM'
df_count = df_count[df_count['Methodologies Used']!='AR-']
df_count['Methodologies Used'] = df_count['Methodologies Used'].replace(['AM0'],['AM'])
df_count['Methodologies Used'].value_counts()

ACM    4330
AMS    3254
AM      520
Name: Methodologies Used, dtype: int64

In order to clearly classify the industries involved in each project with ACM, AMS, or AM, we revise the name whose first three words are 'AMO' to 'AM' and delete the projects starting with 'AR-'.

In [66]:
df_value_counts =df_count.value_counts()

df_value_counts = df_value_counts.reset_index()
df_value_counts.columns = ['Sectoral scopes','Methodologies Used', 'counts']
df_value_counts

Sectoral scopes Methodologies Used  \
0   1 : Energy industries (renewable - / non-renew...                ACM   
1   1 : Energy industries (renewable - / non-renew...                AMS   
2   1 : Energy industries (renewable - / non-renew...                AMS   
3                    13 : Waste handling and disposal                AMS   
4   1 : Energy industries (renewable - / non-renew...                ACM   
5                    13 : Waste handling and disposal                ACM   
6   1 : Energy industries (renewable - / non-renew...                 AM   
7                                   3 : Energy demand                AMS   
8                        4 : Manufacturing industries                AMS   
9   8 : Mining/mineral production\n10 : Fugitive e...                ACM   
10                            5 : Chemical industries                 AM   
11    3 : Energy demand\n4 : Manufacturing industries                AMS   
12  1 : Energy industries (renewable - / non-renew...                 AM   
13                       4 : Manufacturing industries                ACM   
14  13 : Waste handling and disposal\n15 : Agricul...                 AM   
15  10 : Fugitive emissions from fuels (solid, oil...                 AM   
16  1 : Energy industries (renewable - / non-renew...                 AM   
17                   13 : Waste handling and disposal                 AM   
18  1 : Energy industries (renewable - / non-renew...                AMS   
19  1 : Energy industries (renewable - / non-renew...                ACM   
20  1 : Energy industries (renewable - / non-renew...                AMS   
21                            5 : Chemical industries                ACM   
22  11 : Fugitive emissions from production and co...                 AM   
23                       4 : Manufacturing industries                 AM   
24  1 : Energy industries (renewable - / non-renew...                AMS   
25                               9 : Metal production                 AM   
26                                  3 : Energy demand                 AM   
27                                      7 : Transport                ACM   
28                                      7 : Transport                AMS   
29  8 : Mining/mineral production\n1 : Energy indu...                ACM   
30  13 : Waste handling and disposal\n15 : Agricul...                ACM   
31  1 : Energy industries (renewable - / non-renew...                AMS   
32                                      7 : Transport                 AM   
33                            2 : Energy distribution                AMS   
34  13 : Waste handling and disposal\n15 : Agricul...                AMS   
35  11 : Fugitive emissions from production and co...                 AM   
36  10 : Fugitive emissions from fuels (solid, oil...                 AM   
37                            5 : Chemical industries                AMS   
38    13 : Waste handling and disposal\n7 : Transport                AMS   
39  1 : Energy industries (renewable - / non-renew...                 AM   
40  1 : Energy industries (renewable - / non-renew...                AMS   
41  1 : Energy industries (renewable - / non-renew...                ACM   
42  1 : Energy industries (renewable - / non-renew...                ACM   
43  9 : Metal production\n1 : Energy industries (r...                 AM   
44  10 : Fugitive emissions from fuels (solid, oil...                AMS   
45  1 : Energy industries (renewable - / non-renew...                 AM   
46  5 : Chemical industries\n13 : Waste handling a...                AMS   
47  4 : Manufacturing industries\n13 : Waste handl...                 AM   
48  1 : Energy industries (renewable - / non-renew...                ACM   
49  13 : Waste handling and disposal\n5 : Chemical...                AMS   
50  1 : Energy industries (renewable - / non-renew...                 AM   
51  9 : Metal production\n11 : Fugitive emissions ...                 AM   

    count

We use value_counts() to get the number of projects for different CDM scopes with ACM, AMS, or AM and set the index number to view more clearly.

### Export to an excel file

In [64]:
df_value_counts.to_excel('number of project with different methodologies.xlsx', sheet_name='sheet1', index=False)

Then we export the dataframe to the excel file. For the detailed classification of projects in different industry scopes and methodologies, please refer to the file 'number of project with different methodologies'.